helix-vcs/src/lib.rs

```diff
@@ -0,0 +1,49 @@
+mod git;
+mod rope;
+
+use std::{
+    cell::RefCell,
+    collections::HashMap,
+    path::{Path, PathBuf},
+    rc::Rc,
+};
+
+pub use git::Git;
+
+// TODO: Move to helix_core once we have a generic diff mode
+#[derive(Copy, Clone, Debug)]
+pub enum LineDiff {
+    Added,
+    Deleted,
+    Modified,
+}
+
+/// Maps line numbers to changes
+pub type LineDiffs = HashMap<usize, LineDiff>;
+
+pub type RepoRoot = PathBuf;
+
+#[derive(Debug, Default)]
+pub struct Registry {
+    inner: HashMap<RepoRoot, Rc<RefCell<Git>>>,
```

Comment:
Not sure if it's a good idea to have this as `Git` given that it is called vcs, I thought it is supposed to be extensible to other vcs.